File: server/src/main/java/ru/aim/pilot/ui/RevisionForm.java
package ru.aim.pilot.ui;

import com.vaadin.ui.Component;
import com.vaadin.ui.TextField;
import org.vaadin.viritin.fields.IntegerField;
import org.vaadin.viritin.fields.MTextArea;
import org.vaadin.viritin.fields.MTextField;
import org.vaadin.viritin.form.AbstractForm;
import org.vaadin.viritin.layouts.MFormLayout;
import org.vaadin.viritin.layouts.MVerticalLayout;
import ru.aim.pilot.model.Revision;

public class RevisionForm extends AbstractForm<Revision> {

    private TextField subjectName = new MTextField("Наименование субъекта Российской Федерации");
    private TextField address = new MTextField("Наименование ОПО, ГТС, наименование, адрес, ИНН эксплуатирующей организации");
    private TextField inn = new MTextField("ИНН организации");
    private IntegerField checkCount = new IntegerField().withCaption("Количество проверок");
    private IntegerField allViolationsCount = new IntegerField().withCaption("Общее число нарушений");
    private IntegerField fixedViolationsCount = new IntegerField().withCaption("Число устранённых нарушений");
    private MTextArea violationsDesc = new MTextArea("Выявленные нарушения проверяемых систем, режима и охраны");
    private MTextArea violationsMark = new MTextArea("Отметка об устранении нарушений");

    public RevisionForm() {
        setSizeUndefined();
    }

    @Override
    protected Component createContent() {
        return new MVerticalLayout(
                new MFormLayout(
                        subjectName,
                        address,
                        inn,
                        checkCount,
                        allViolationsCount,
                        fixedViolationsCount,
                        violationsDesc,
                        violationsMark
                ).withWidth(""),
                getToolbar()
        ).withWidth("");
    }

    @Override
    public String getCancelCaption() {
        return "Отмена";
    }

    @Override
    public String getSaveCaption() {
        return "Сохранить";
    }
}

File: server/src/main/java/ru/aim/pilot/ui/RevisionForm.kt
package ru.aim.pilot.ui

import com.vaadin.ui.Component
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Scope
import org.vaadin.viritin.fields.IntegerField
import org.vaadin.viritin.fields.MTextArea
import org.vaadin.viritin.fields.MTextField
import org.vaadin.viritin.form.AbstractForm
import org.vaadin.viritin.layouts.MFormLayout
import org.vaadin.viritin.layouts.MVerticalLayout
import ru.aim.pilot.model.Revision
import ru.aim.pilot.spring.UiStringResolver
import org.springframework.stereotype.Component as SpringComponent

@SpringComponent
@Scope("prototype")
open class RevisionForm
@Autowired
constructor(private val resolver: UiStringResolver) : AbstractForm<Revision>() {

    private val subjectName = MTextField(Revision::subjectName.name)
    private val address = MTextArea(Revision::address.name)
    private val inn = MTextField(Revision::inn.name)
    private val typeSafeSystem = MTextArea(Revision::typeSafeSystem.name)
    private val checkCount = IntegerField().withCaption(Revision::checkCount.name)
    private val allViolationsCount = IntegerField().withCaption(Revision::allViolationsCount.name)
    private val fixedViolationsCount = IntegerField().withCaption(Revision::fixedViolationsCount.name)
    private val violationsDesc = MTextArea(Revision::violationsDesc.name)
    private val violationsMark = MTextArea(Revision::violationsMark.name)

    init {
        setSizeUndefined()
    }

    override fun createContent(): Component {
        val formLayout = MFormLayout(
                subjectName,
                address,
                inn,
                typeSafeSystem,
                checkCount,
                allViolationsCount,
                fixedViolationsCount,
                violationsDesc,
                violationsMark).withWidth("")
        formLayout.forEach { it.caption = resolver.resolveKey(it.caption) }
        return MVerticalLayout(
                formLayout,
                toolbar).withWidth("")
    }

    override fun getCancelCaption(): String {
        return resolver.resolveKey("cancel")
    }

    override fun getSaveCaption(): String {
        return resolver.resolveKey("save")
    }
}